extensions/registration-service/src/main/java/org/eclipse/dataspaceconnector/registration/AuthorityExtension.java

```diff
@@ -129,8 +130,8 @@ public ParticipantStore participantStore() {
     }
 
     @Provider(isDefault = true)
-    public CredentialsVerifier credentialsVerifier() {
-        return new DummyCredentialsVerifier();
+    public ParticipantVerifier participantVerifier(ServiceExtensionContext context) {
+        return new DefaultParticipantVerifier(context.getMonitor(), context.getService(DidResolverRegistry.class), context.getService(CredentialsVerifier.class));
     }
```

Comment:
Are we sure this wont lead to Dependency Resolution issues? 

According to what we described with Paul in the doc here https://github.com/eclipse-dataspaceconnector/DataSpaceConnector/blob/main/docs/developer/dependency_resolution.md#provide-defaults this does not seem right, as `DidResolverRegistry` and `CredentialsVerifier` might not have been initialized yet when this default provider is called.

Instead I would see something like this:

~~~
@Inject(required = false)
private ParticipantVerifier participantVerifier;

public void initialize(ServiceExtensionContext context) {
   ...
   if(participantVerifier == null) {
      context.registerService(ParticipantVerifier.class, new DefaultParticipantVerifier(...));
   }
}
~~~

Reply:
We do have `@Requires({PrivateKeyResolver.class, OkHttpClient.class, DidResolverRegistry.class, CredentialsVerifier.class})`, so we are guaranteed that the collaborators are initialised before this is called.

Besides, how would your suggested fix help if DidResolverRegistry and CredentialsVerifier were not initialised for some unexpected reason?

Reply:
> We do have `@Requires({PrivateKeyResolver.class, OkHttpClient.class, DidResolverRegistry.class, CredentialsVerifier.class})`, so we are guaranteed that the collaborators are initialised before this is called.
> 

I am pretty sure that this statement is only true for what is located in the `initialize` method. 

For my proposed fix, this would look like this:

~~~
@Inject(required = false)
private DidResolverRegistry didResolverRegistry;

@Inject(required = false)
private CredentialsVerifier credentialsVerifier;

public void initialize(ServiceExtensionContext context) {
   ...
   if(participantVerifier == null) {
      Objects.requireNotNull(didResolverRegistry, "DidResolverRegistry is mandatory for creating DefaultParticipantVerifier");
      Objects.requireNotNull(credentialsVerifier, "CredentialsVerifier is mandatory for creating DefaultParticipantVerifier");
      context.registerService(ParticipantVerifier.class, new DefaultParticipantVerifier(context.getMonitor(), didResolverRegistry, credentialsVerifier));
   }
}
~~~

Reply:
the problem that @bscholtes1A points out is that the default provider methods are typically called during the `inject` phase of the extension lifecycle. Here, it will work, more by chance than anything else, because the corresponding `@Inject` is coming from the same extension, but as soon as that is in a different extension it may not work anymore, because `@Provider` methods's dependencies are not yet considered by the DI mechanism. 

In other words, the `@Requires` and `@Provides` annotations are not considered when invoking a default provider.

For now, the safest way is to move the provider method into its own extension. I'd also remove the `isDefault` annotation - it was never intended to achieve extensibility. That's what extensions are for.

Reply:
Separated extension